src/other/index.jl

```diff
@@ -112,7 +112,10 @@ Base.getindex(x::Index, idx::Symbol) = x.lookup[idx]
 Base.getindex(x::AbstractIndex, idx::Real) = Int(idx)
 Base.getindex(x::AbstractIndex, idx::AbstractVector{Union{Bool, Missing}}) =
     getindex(x, collect(Missings.replace(idx, false)))
-Base.getindex(x::AbstractIndex, idx::AbstractVector{Bool}) = find(idx)
+function Base.getindex(x::AbstractIndex, idx::AbstractVector{Bool})
+    length(x) == length(idx) || Base.throw_boundserror(x, idx)
```

Comment:
I'd rather not call an internal Base function, especially since it's so simple. If you're afraid of stopping the compiler from inlining this method, just copy the definition here.